virttest/iscsi.py

```diff
@@ -833,5 +835,494 @@ def create_iSCSI(params, root_dir=data_dir.get_tmp_dir()):
                 )
             iscsi_instance = IscsiTGT(params, root_dir)
         else:
-            iscsi_instance = IscsiLIO(params, root_dir)
+            iscsi_instance = (
+                MultiPathsIscsiLIOManager(params, root_dir)
+                if not params.get("target")
+                and params.get("iscsi_targets", "")
+                and params.get("iscsi_target_backstores", "")
+                else IscsiLIO(params, root_dir)
+            )
         return iscsi_instance
+
+
+class Backstore(object):
+    iscsi_backend = None
+
+    def __init__(self):
+        pass
+
+    def create_backstore(self):
+        raise NotImplementedError
+
+    def delete_backstore(self):
+        raise NotImplementedError
+
+
+class Fileio(Backstore):
+    """
+    Model of the fileio backend in backstore.
+    """
+
+    iscsi_backend = "fileio"
+
+    def __init__(self, params, root_dir):
+        """
+        Initialize LIO backend of backstore.
+
+        :param params: parameters for LIO backend of backstore.
+        :type params: dict
+        :param root_dir: path for image
+        :type root_dir: string
+        """
+        self.initiator = None
+        self.iscsi_lun_attrs = params.get("attrs")
+        emulated_image = params.get("emulated_image")
+        self.emulated_image = (
+            emulated_image
+            if os.path.isabs(emulated_image)
+            else os.path.join(root_dir, emulated_image)
+        )
+        self.backstore_name = params.get("backstore_name")
+        self.emulated_id = ""
+        self.emulated_size = params.get("emulated_image_size")
+        self.unit = self.emulated_size[-1].upper()
+        self.emulated_size = self.emulated_size[:-1]
+        # maps K,M,G,T => (count, bs)
+        emulated_size = {
+            "K": (1, 1),
+            "M": (1, 1024),
+            "G": (1024, 1024),
+            "T": (1024, 1048576),
+        }
+        if self.unit in emulated_size:
+            block_size = emulated_size[self.unit][1]
+            size = int(self.emulated_size) * emulated_size[self.unit][0]
+            self.emulated_expect_size = block_size * size
+            self.create_cmd = "dd if=/dev/zero of=%s count=%s bs=%sK" % (
+                self.emulated_image,
+                size,
+                block_size,
+            )
+        else:
+            raise exceptions.TestError(
+                "Image size provided is not in valid"
+                " format, specify proper units [K|M|G|T]"
```

Comment:
So we also need to update base.cfg with the comments: "emulated_image_size = xx<K|M|G|T> "